core/common/src/main/java/alluxio/heartbeat/SleepingTimer.java

```diff
@@ -26,51 +26,48 @@
  * This class can be used for executing heartbeats periodically.
  */
 @NotThreadSafe
-public final class SleepingTimer implements HeartbeatTimer {
+public class SleepingTimer implements HeartbeatTimer {
   private long mIntervalMs;
-  private long mPreviousTickMs;
+  protected long mPreviousTickMs = -1;
   private final String mThreadName;
-  private final Logger mLogger;
-  private final Clock mClock;
-  private final Sleeper mSleeper;
+  protected final Logger mLogger;
+  protected final Clock mClock;
+  protected final Sleeper mSleeper;
+  protected final Supplier<Long> mIntervalSupplier;
```

Comment:
Can we implement an adapter that turns a `CronExpression` into a `Supplier<Long>`?